contracts/VaultV3.vy

```diff
@@ -612,7 +612,7 @@ def _revoke_strategy(old_strategy: address):
 @internal
 def _migrate_strategy(new_strategy: address, old_strategy: address, call_migrate_strategy: bool):
     assert self.strategies[old_strategy].activation != 0, "old strategy not active"
-    assert self.strategies[old_strategy].current_debt == 0, "old strategy has debt"
+    assert self.strategies[old_strategy].current_debt == 0 or call_migrate_strategy, "old strategy has debt"
     assert new_strategy != empty(address), "strategy cannot be zero address"
     assert IStrategy(new_strategy).asset() == ASSET.address, "invalid asset"
     assert IStrategy(new_strategy).vault() == self, "invalid vault"
```

Comment:
this check for .vault() means that it wont be compatible with other 4626 non-yearn deployed contract. Is that the intention?

Reply:
will address in issue #128